core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -61,6 +61,14 @@ sealed trait TaskFailedReason extends TaskEndReason {
    * on was killed.
    */
   def countTowardsTaskFailures: Boolean = true
+
+  /**
+   * Whether this task failure should be counted towards the maximum number of times the stage is
+   * allowed to fail before the stage is aborted. Set to false in cases where the task's failure
+   * was unrelated to the task; for example, if the task failed because fetch failed exception
+   * from the decommissioned node.
+   */
+  var countTowardsStageFailures: Boolean = true
```

Comment:
Why is this a var?

Reply:
We updated the value of countTowardsStageFailures when the 

```
// Do account fetch failure exception raised by decommissioned
          // node against stage failure. Here the logic is to specify,
          // if the task failed due to fetchFailed of decommission nodes than
          // don't count towards the stageFailure. countTowardsStageFailures
          // variable of TaskEndReason, that can be used in DAG scheduler to account
          // fetch failure while checking the stage abort
          decommissionTracker match {
            case Some(decommissionTracker) =>
              if (decommissionTracker.isNodeDecommissioned(fetchFailed.bmAddress.host)) {
                logInfo(s"Do not count fetch failure from decommissioned" +
                  s" node ${fetchFailed.bmAddress.host}")
                fetchFailed.countTowardsStageFailures = false
              }
            case _ =>
            // No action
          }
```